src/rust/src/pkcs7.rs

```diff
@@ -95,14 +95,19 @@ fn encrypt_and_serialize<'p>(
         smime_canonicalize(raw_data.as_bytes(), text_mode).0
     };
 
-    // The message is encrypted with AES-128-CBC, which the S/MIME v3.2 RFC
-    // specifies as MUST support (https://datatracker.ietf.org/doc/html/rfc5751#section-2.7)
-    let key = types::OS_URANDOM.get(py)?.call1((16,))?;
-    let aes128_algorithm = types::AES128.get(py)?.call1((&key,))?;
+    // Get the algorithm
+    let algorithm_type = builder.getattr(pyo3::intern!(py, "_algorithm"))?;
+    let key_size = algorithm_type.getattr(pyo3::intern!(py, "key_size"))?;
```

Comment:
This is wrong for many `type[CipherAlgorithm]`, it really onyl works with AES128 and AES256.

Reply:
That's correct, blocking it in the Python side!

Reply:
Simplified the way it works to only handle the 2 algorithm types.